frontend/src/console/components/AddDomainButton/AddDomainButton.tsx

```diff
@@ -0,0 +1,269 @@
+import { InstancesModel } from 'console/models';
```

Comment:
@arjunsinghy96 This component is doing way too much than its name is suggesting. Can we break it down into a couple of pieces or Change the name to a more appropriate one?

This is not really a button component anymore.

Reply:
@shimulch This makes sense. I think `AddDomainModalButton` or simple `AddDomain` makes sense.

Reply:
I went ahead with `AddDomainModalButton` as it felt sufficiently verbose.